src/libcharon/plugins/kernel_super/oper_db_client/liboper_db_client/liboper_db_client.h

```diff
@@ -7,36 +7,87 @@ extern "C" {
     LIBOPER_DB_CLIENT_CLASS_ID_TERMINATOR,
-    LIBOPER_DB_CLIENT_CLASS_ID_VRF_TUNNELS_NUMBER_IN_TERMINATOR
+    LIBOPER_DB_CLIENT_CLASS_ID_VRF_TUNNELS_NUMBER_IN_TERMINATOR,
+    LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_ITEM,
+    LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA,
 };
 
 enum LIBOPER_DB_CLIENT_FIELD_IDS {
     LIBOPER_DB_CLIENT_FIELD_ID_TERMINATOR_OPER_STATE,
     LIBOPER_DB_CLIENT_FIELD_ID_TERMINATOR_TUNNELS_NUMBER,
-    LIBOPER_DB_CLIENT_FIELD_ID_VRF_TERMINATOR_TUNNELS_NUMBER
+    LIBOPER_DB_CLIENT_FIELD_ID_VRF_TERMINATOR_TUNNELS_NUMBER,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_VRF_NAME,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_ID,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_TERMINATOR_ID,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_STATUS,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ACTIVENESS_STATUS,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ACCOUNT_ID,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_VERSION,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_NEIGHBOR_SUB_INTERFACE,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DH_GROUP,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ENCRYPTION_ALGO,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INTEGRITY_ALGO,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INBOUND_IPSEC_SPI,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_OUTBOUND_IPSEC_SPI,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INBOUND_IKE_SPI,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_OUTBOUND_IKE_SPI,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_NAT_IDENTIFIED,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_TUNNEL_UPTIME,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ANTI_REPLAY_WINDOW_SIZE,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_EXT_SEQ_NUM,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_PUBLIC_ADDRESS,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_REMOTE_PUBLIC_ADDRESS,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_PUBLIC_SRC_PORT,
+    LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_LOCAL_NETWORK,
 };
 
 enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES {
     LIBOPER_DB_CLIENT_STATE_ENUM_UP,
     LIBOPER_DB_CLIENT_STATE_ENUM_DOWN,
+    LIBOPER_DB_CLIENT_STATUS_IKE_CREATED,
+    LIBOPER_DB_CLIENT_STATUS_IKE_CONNECTING,
+    LIBOPER_DB_CLIENT_STATUS_IKE_ESTABLISHED,
+    LIBOPER_DB_CLIENT_STATUS_IKE_REKEYING,
+    LIBOPER_DB_CLIENT_STATUS_IKE_REKEYED,
+    LIBOPER_DB_CLIENT_STATUS_IKE_DELETING,
+    LIBOPER_DB_CLIENT_ACTIVENESS_STATUS_ACTIVE,
+    LIBOPER_DB_CLIENT_ACTIVENESS_STATUS_STANDBY,
+    LIBOPER_DB_CLIENT_IKE_VERSION_1,
+    LIBOPER_DB_CLIENT_IKE_VERSION_2,
+    LIBOPER_DB_CLIENT_ENC_ALG_AES_128_CBC,
+    LIBOPER_DB_CLIENT_ENC_ALG_AES_256_CBC,
+    LIBOPER_DB_CLIENT_INT_ALG_SHA2_256,
+    LIBOPER_DB_CLIENT_INT_ALG_SHA1,
 };
 
+enum LIBOPER_DB_CLIENT_UNION_FIELD_ID {
+    LIBOPER_DB_CLIENT_IPV4_ADDRESS,
+};
 
 struct liboper_db_client_init_params {
     const char* proc_name;
 };
 
+struct liboper_union_str
+{
+	union
```

Comment:
No point in adding the union. `str` can just be a "regular" field in the struct together with `field_id`